src/lib.rs

```diff
@@ -220,10 +223,27 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
-    } else if #[cfg(any(target_os = "android", target_os = "linux"))] {
+    } else if #[cfg(all(
+        not(feature = "disable_urandom_fallback"),
+        target_os = "linux",
+        any(
+            target_arch = "i586",
+            target_arch = "i686",
+            target_arch = "x86",
+            target_arch = "x86_64",
+            target_arch = "arm",
+            target_arch = "armv7",
+            target_arch = "powerpc",
+            target_arch = "powerpc64le",
+            target_arch = "s390x",
+        ),
+    ))] {
```

Comment:
Note that the link I suggest above is for the Rust 1.47.0 platform support documentation. The MSRV of getrandom is older than 1.47.0, but 1.47.0 is the oldest version that has documentation online that I could find. Perhaps it would be better to link to the source of the file in GitHub as of whatever revision was in effect for our MSRV.

Reply:
I've looked into it a bit more and it looks like Rust supports Android API level 19 (KitKat), while `getrandom(2)` was added only in level 23 (Marshmallow). So it's too soon to disable the file fallback for Android.